llvm/tools/llvm-cas/llvm-cas.cpp

```diff
@@ -522,21 +553,25 @@ Error checkCASIngestPath(StringRef CASPath, StringRef DataPath) {
 int ingestFileSystem(ObjectStore &CAS, std::optional<StringRef> CASPath,
                      StringRef Path) {
   ExitOnError ExitOnErr("llvm-cas: ingest: ");
-  if (Path.empty())
-    ExitOnErr(
-        createStringError(inconvertibleErrorCode(), "missing --data=<path>"));
-  if (CASPath)
+  if (CASIDFile) {
+    if (!Path.empty())
+      ExitOnErr(createStringError(
+          inconvertibleErrorCode(),
+          "incorrect usage of --data=<path> with --casid-file"));
+    ingestFile(CAS);
+  }
+  if (CASPath && !Path.empty())
```

Comment:
I think there is value in having an option where `llvm-cas` can ingest all files in a given directory, so I added support for positional arguments, without removing support for the `--data` option

Reply:
I would argue that it really doesn't matter if the input is directory or file, you just do the same thing.

All you need is update `recursiveAccess` that:
* Visit file and add to HierarchicalTreeBuilder with the correct file type
* If casid-file is passed, treat all the file visited as CASID file
Then you build the tree out of that.

I don't really like the `--data` option right now, it is not very obvious what is going on in there. If you don't want to update all the tests, I guess you can just merge `--data` arguments and positional arguments.